core/resource/resource.go

```diff
@@ -67,6 +68,8 @@ type Filter struct {
 }
 
 type YAML struct {
```

Comment:
i know it's not part of this change. but this struct name doesn't seem right. the fields are tagged with json & yaml tags. so it's not dedicated for yaml generation. and the name doesn't provide enough context

Reply:
I am not sure where do we use the `json` tags is. But AFAIK this struct is used to parse yaml file and populated from it.

cc @krtkvrm

Reply:
hmm, i think it's okay to leave the json tags in the struct. but i feel the name could be better (maybe based on what data it represents or something like that)

Reply:
@spy16 any suggestion on that?
Since this is just a placeholder for yaml data, I was thinking something like `Config`.

Reply:
isn't this the resource-type definition from the file? 🤔 `ResourceTypeDef` could be used in that case?